velox/exec/tests/ExpandTest.cpp

```diff
@@ -0,0 +1,218 @@
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY ROLLUP (k1, k2)");
+}
+
+TEST_F(ExpandTest, countDistinct1) {
+  vector_size_t size = 5;
+  auto data = makeRowVector(
+      {"a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<std::string>(
+              size, [](auto row) { return std::string(row % 12, 'x'); }),
+      });
+  createDuckDbTable({data});
+  std::cout << "the input data is " << data->toString(0, 5) << std::flush
+            << std::endl;
+
+  // count distinct.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand({{"a", "1 as group_id_0"}, {"null", "2"}})
+          .singleAggregation({"group_id_0"}, {"count(distinct a) as count_a"})
+          .project({"count_a"})
+          .planNode();
+
+  assertQuery(plan, "SELECT count(distinct a) FROM tmp");
+}
+
+TEST_F(ExpandTest, countDistinct) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<std::string>(
+              size, [](auto row) { return std::string(row % 12, 'x'); }),
+      });
+  createDuckDbTable({data});
+
+  // count distinct.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand({{"a", "null as b", "1 as group_id_0"}, {"null", "b", "2"}})
+          .singleAggregation(
+              //   {"group_id_0"},
+              {},
+              {"count(distinct a) as count_a", "count(distinct b) as count_b"})
+          .project({"count_a", "count_b"})
+          .planNode();
+
+  assertQuery(plan, "SELECT count(distinct a), count(distinct b) FROM tmp");
+}
+
+// TEST_F(ExpandTest, duplicatedName) {
+//   vector_size_t size = 1'000;
+//   auto data = makeRowVector(
+//       {"k1", "k2", "a", "b"},
+//       {
+//           makeFlatVector<int64_t>(size, [](auto row) { return row % 11; }),
+//           makeFlatVector<int64_t>(size, [](auto row) { return row % 17; }),
+//           makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+//           makeFlatVector<std::string>(
+//               size, [](auto row) { return std::string(row % 12, 'x'); }),
+//       });
+
+//   createDuckDbTable({data});
+
+//   auto plan =
+//       PlanBuilder()
+//           .values({data})
+//           .expand(
+//               {{"k1", "k1", "a", "b", "0 as group_id_0"},
+//                {"k1", "null", "a", "b", "1"},
+//                {"null", "null", "a", "b", "2"}})
+//           .singleAggregation(
+//               {"k1", "k2", "group_id_0"},
+//               {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+//           .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+//           .planNode();
+
+//   VELOX_ASSERT_USER_THROW(
```

Comment:
@mbasmanova I try to add this test to check the duplicated names. But i got the following exception when running this test. Do you have any input? Thanks.

```
unknown file: Failure
C++ exception with description "Exception: VeloxUserError
Error Source: USER
Error Code: INVALID_ARGUMENT
Reason: Found duplicate column name in Expand plan node: k1.
Retriable: False
Expression: result.second
Function: getExpandOutputType
File: ../../velox/core/PlanNode.cpp
Line: 467
Stack trace:
Stack trace has been disabled. Use --velox_exception_user_stacktrace_enabled=true to enable it.
" thrown in the test body.

```

Reply:
@JkSelf Not sure I understand the question. Isn't this exception expected?

Reply:
Since exception occurs on PlanNode creation, the test should expect PlanBuilder::expand call to fail and wrap it in VELOX_ASSERT_USER_THROW macro.

Reply:
@mbasmanova Yes, this exception should be caught, but it seems that it hasn't been caught.

Reply:
@JkSelf Try putting VELOX_ASSERT_THROW around PlanBuilder::expand call.

```
VELOX_ASSERT_THROW(
     PlanBuilder()
         .values({data})
         .expand(..), ...);
```